core/domain/rights_manager.py

```diff
@@ -1111,11 +1017,25 @@ def _unpublish_activity(committer_id, activity_id, activity_type):
         Exception. The committer does not have rights to unpublish the
             activity.
     """
-    if not Actor(committer_id).can_unpublish(activity_type, activity_id):
-        logging.error(
-            'User %s tried to unpublish %s %s but was refused '
-            'permission.' % (committer_id, activity_type, activity_id))
-        raise Exception('This %s cannot be unpublished.' % activity_type)
+    committer_role = user_services.get_user_role_from_id(committer_id)
+    committer_actions = role_services.get_all_actions(committer_role)
+    activity_rights = _get_activity_rights(activity_type, activity_id)
+
+    if activity_type == constants.ACTIVITY_TYPE_EXPLORATION:
+        if not check_can_unpublish_exploration(
+                committer_actions, activity_rights):
+            logging.error(
+                'User %s tried to unpublish %s %s but was refused '
+                'permission.' % (committer_id, activity_type, activity_id))
+            raise Exception('This %s cannot be unpublished.' % activity_type)
+
+    if activity_type == constants.ACTIVITY_TYPE_COLLECTION:
+        if not check_can_unpublish_collection(
```

Comment:
Just wondering: why do we have separate blocks for explorations and collections here?